### PR TITLE
Add RESTful to WMTS GetCapabilities optionsFromCapabilities

### PR DESCRIPTION
**Pull request by sarametz**
The WMTS optionsFromCapabilities PR (#3142) was merged without allowing RESTful as a valid requestEncoding parameter. This PR adds it back in

### CODE REVIEW

**Comment by ahocevar** [comment]
Thanks Sara!

File: src/ol/source/wmtssource.js [hunk 1]
```diff
@@ -350,8 +351,23 @@ ol.source.WMTS.optionsFromCapabilities = function(wmtsCap, config) {
   /** @type {!Array.<string>} */
   var urls = [];
   var requestEncoding = config['requestEncoding'];
-  if (wmtsCap['OperationsMetadata'].hasOwnProperty('GetTile') &&
-      requestEncoding != 'REST') {
+  if (!requestEncoding) {
+    requestEncoding = '';
+  }
+  if (!goog.array.contains(['REST', 'RESTful', 'KVP', ''], requestEncoding)) {
```
Comment (ahocevar) [inline src/ol/source/wmtssource.js]: Could be simplified as
```js
goog.asserts(goog.array.contains(['REST', 'RESTful', 'KVP', ''], requestEncoding));
```

File: src/ol/source/wmtssource.js [hunk 2]
```diff
@@ -350,8 +351,23 @@ ol.source.WMTS.optionsFromCapabilities = function(wmtsCap, config) {
...
+    goog.asserts.fail();
+  }
+  if (!wmtsCap['OperationsMetadata'].hasOwnProperty('GetTile') ||
+      goog.string.startsWith(requestEncoding, 'REST')) {
+    // Add REST tile resource url
+    requestEncoding = ol.source.WMTSRequestEncoding.REST;
+    goog.array.forEach(l['ResourceURL'], function(elt, index, array) {
```
Comment (ahocevar) [inline src/ol/source/wmtssource.js]: We usually use plain JavaScript `for` loops instead of `goog.array.forEach`.

**Comment by ahocevar** [comment]
Looks good. I only have one minor comment about simplifying an assertion.

**Comment by sarametz** [comment]
Thanks @ahocevar, I have updated the PR with the change you suggested.

File: src/ol/source/wmtssource.js [hunk 3]
```diff
@@ -350,8 +351,23 @@ ol.source.WMTS.optionsFromCapabilities = function(wmtsCap, config) {
   /** @type {!Array.<string>} */
   var urls = [];
   var requestEncoding = config['requestEncoding'];
-  if (wmtsCap['OperationsMetadata'].hasOwnProperty('GetTile') &&
-      requestEncoding != 'REST') {
+  if (!requestEncoding) {
```
Comment (elemoine) [inline src/ol/source/wmtssource.js]: Please use `goog.isDef` if you want to test whether `requestEncoding` is defined.

**Comment by sarametz** [comment]
The PR has been updated to use `goog.isDef` as @elemoine suggested. Let me know if any more changes are required

**Comment by ahocevar** [comment]
Thanks @sarametz for this valuable contribution!

**Pull request merged**